test/assertions.js

```diff
@@ -1,8 +1,48 @@
 import should from 'should';
 
+import { wait } from './util/util';
+
 const unwrapElement = (elementOrWrapper) => (elementOrWrapper instanceof HTMLElement
   ? elementOrWrapper
   : elementOrWrapper.element);
+
+// Assertion.addSync() is like Assertion.add(), but the specified function can
+// be async.
+const { Assertion, AssertionError } = should;
+Assertion.addAsync = (name, f) => {
```

Comment:
Should.js has async assertions which are `await`ed, for example:

```js
await somePromise.should.be.resolved();
await anotherPromise.should.be.rejectedWith('some error');
```

However, while Should.js provides a way to add custom synchronous assertions, I don't see a way to add custom async assertions. The tooltip assertions need to be async because `computePosition()` from Floating UI is async. I added this method as a way to add custom async assertions. If the assertion fails, a useful Should.js `AssertionError` is thrown.